src/components/select/Select.js

```diff
@@ -738,7 +738,7 @@ export default class SelectComponent extends BaseComponent {
 
     // Add value options.
     if (this.addValueOptions()) {
-      this.restoreValue();
+      this.choices.setChoiceByValue(this.dataValue);
```

Comment:
This will now break if this.choices is not defined (like when using the HTML5 widget)